### PR TITLE
Add diagrams 🎨

### PR DESCRIPTION
**Pull request by matthewjablack**
This PR adds diagrams showing an overview of on-chain and off-chain transactions involved with entering and exiting a DLC. 

It also includes a fee-bumping diagram and depiction of participant/outsider view of a DLC from a block explorer. 

Thanks to @Abhimanyu1994 for putting these together 🙏 🔥  and @nkohen for the initial inspiration

**Overview:**

dlc_overview

**Overview + RBF:**

dlc_overview_rbf

**Participant View:**

dlc_participant_view

**Outsider View:**

dlc_outsider_view

### CODE REVIEW

File: Introduction.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ by every party involved.
 DLC [were originally proposed by Thaddeus Dryja](https://adiabat.github.io/dlc.pdf).
 The goal of DLC is to enable setting up contracts between two parties directly on the Bitcoin blockchain using an oracle to determine the contract outcome.
 
+![DLC Overview](/images/dlc_overview.png)
+
```
Comment (ghost) [inline Introduction.md:13]: There is a typo in this image. It should be `UTXOs` instead of `UXTOS`

**Review by ghost — CHANGES_REQUESTED** [review]
There is only one typo. Everything else looks good and will be helpful.

File: Introduction.md (hunk 1 above)
Comment (benthecarman) [reply Introduction.md:13]: In `dlc_overview_rbf` as well

Comment (matthewjablack) [reply Introduction.md:13]: Good catch, updated

Comment (ghost) [reply Introduction.md:13]: It's still incorrect in `dlc_overview`. `X` is at wrong place. 

Incorrect: U**X**TOs
Correct: U**TX**Os

Comment (matthewjablack) [reply Introduction.md:13]: 2nd time is the charm haha. Updated thanks

File: Introduction.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ by every party involved.
 DLC [were originally proposed by Thaddeus Dryja](https://adiabat.github.io/dlc.pdf).
 The goal of DLC is to enable setting up contracts between two parties directly on the Bitcoin blockchain using an oracle to determine the contract outcome.
 
+![DLC Overview](/images/dlc_overview.png)
```
Comment (nkohen) [inline Introduction.md]: For me (locally, not on github) these images only get properly linked in without the `/` prefix which I don't think should be here (on all pic links)

Comment (matthewjablack) [reply Introduction.md]: Works either way for me so I'll switch to no `/` prefix.

**Review by nkohen — APPROVED** [review]

**Pull request merged**